install.psm1

```diff
@@ -0,0 +1,264 @@
+﻿#
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#Requires -RunAsAdministrator
+
+function Get-Install-Directory([string]$InstallDir) {
+    $dir = "OpenTelemetry .NET AutoInstrumentation"
+    
+    if ($InstallDir -eq "<auto>" -or $installDir -eq "AppData") {
+        return (Join-Path $env:LOCALAPPDATA "Programs" | Join-Path -ChildPath $dir)
+    }
+    elseif ($InstallDir -eq "ProgramFiles") {
+        return (Join-Path $Env:ProgramFiles $dir)
+    } 
+    elseif (Test-Path $InstallDir -IsValid) {
+        return $InstallDir
+    }
+
+    throw "Invalid install directory provided '$InstallDir'"
+}
+
+function Get-Temp-Directory() {
+    return $env:temp
+}
+
+function Prepare-Install-Directory([string]$InstallDir) {
+    New-Item -ItemType Directory -Force -Path $InstallDir | Out-Null
+}
+
+function Download-OpenTelemetry([string]$Version, [string]$Path) {
+    $archive = "opentelemetry-dotnet-instrumentation-windows.zip"
+    $dlUrl = "https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/download/$Version/$archive"
+    $dlPath = Join-Path $Path $archive
+
+    Invoke-WebRequest -Uri $dlUrl -OutFile $dlPath
+
+    return $dlPath
+}
+
+function Extract-OpenTelemetry([string]$DlPath, [string] $InstallPath) {
+    Expand-Archive $DlPath $InstallPath -Force
```

Comment:
Remove `$InstallPath` beforehand to ensure that there are no leftovers from previous installations.

Alternatively, throw an error and ask for calling `Uninstall-OpenTelemetryCore`. However, this may not be the ideal solution if the uninstallation steps change.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1297/commits/ca2ec7ebdc2a96810fd1bb9f389d5ec4109295d7